### PR TITLE
task for creating imagestream for buildroot

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: templates/osbs-buildroot-imagestream.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ImageStream
+metadata:
+  annotations:
+    openshift.io/image.insecureRepository: "true"
+  name: osbs-buildroot
+  namespace: {{ osbs_namespace }}
+spec:
+  dockerImageRepository: brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/rcm/buildroot
```
Comment (vrutkovs) [inline templates/osbs-buildroot-imagestream.yml.j2]: Exposing internal service address is a bad idea.

Also, there is no particular tag to follow

Comment (lcarva) [reply templates/osbs-buildroot-imagestream.yml.j2]: Yeah, this should be a variable. But we don't need to add a tag here. We actually want to tag the whole repo.

File: tasks/orchestrator.yml [hunk 2]
```diff
@@ -16,3 +16,25 @@
...
+  template:
+    src: osbs-buildroot-imagestream.yml.j2
+    dest: "{{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml"
+  environment: "{{ osbs_environment }}"
+  register: yaml_imagestream
+  when: osbs_is_admin
+  tags:
+  - oc
+
+- name: import imagestream
```
Comment (lcarva) [inline tasks/orchestrator.yml]: Let's rename this task. Import imagestream means something else :)

File: templates/osbs-buildroot-imagestream.yml.j2 (hunk 1 above)
Comment (vrutkovs) [reply templates/osbs-buildroot-imagestream.yml.j2]: Every tag in that repo would be imported and a new ImageStreamTag would be created - it would affect the performance.

A better way would be creating a blank ImageStream and tagging latest to a particular buildroot image in the next step. This way we'll have a proper history of buildroots used without extra tags imported

Comment (lcarva) [reply templates/osbs-buildroot-imagestream.yml.j2]: There's a maximum, by default I believe is 5.

Comment (vrutkovs) [reply templates/osbs-buildroot-imagestream.yml.j2]: Ah, correct, its determined by `MaxImagesBulkImportedPerRepository`, which is 5 by default, so its not that important.

Comment (rcerven) [reply templates/osbs-buildroot-imagestream.yml.j2]: ok changed it to variable

File: tasks/orchestrator.yml (hunk 2 above)
Comment (rcerven) [reply tasks/orchestrator.yml]: does create imagestream sounds better?

File: defaults/main.yml [hunk 3]
```diff
@@ -39,6 +39,7 @@ osbs_build_json_dir: /usr/share/osbs
 osbs_sources_command: fedpkg sources
 osbs_vendor: Fedora Project
 osbs_nodeselector: ''
+osbs_builldroot_repository : ''
```
Comment (vrutkovs) [inline defaults/main.yml]: typo

File: tasks/orchestrator.yml [hunk 4]
```diff
@@ -16,3 +16,25 @@
...
   - oc
+
+# Setup imagestream
+- name: copy imagestream
+  template:
+    src: osbs-buildroot-imagestream.yml.j2
+    dest: "{{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml"
+  environment: "{{ osbs_environment }}"
+  register: yaml_imagestream
+  when: osbs_is_admin
```
Comment (twaugh) [inline tasks/orchestrator.yml]: Why is this necessary?

File: templates/osbs-buildroot-imagestream.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ImageStream
+metadata:
+  annotations:
+    openshift.io/image.insecureRepository: "true"
+  name: osbs-buildroot
+  namespace: {{ osbs_namespace }}
+spec:
+  dockerImageRepository: {{ osbs_builldroot_repository }}
```
Comment (twaugh) [inline templates/osbs-buildroot-imagestream.yml.j2]: Typo, should be `osbs_buildroot_repository`

File: tasks/orchestrator.yml (hunk 4 above)
Comment (rcerven) [reply tasks/orchestrator.yml]: probably isn't, will delete it

File: templates/osbs-buildroot-imagestream.yml.j2 (hunk 5 above)
Comment (rcerven) [reply templates/osbs-buildroot-imagestream.yml.j2]: ok

File: defaults/main.yml (hunk 3 above)
Comment (rcerven) [reply defaults/main.yml]: ok

File: templates/osbs-buildroot-imagestream.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ImageStream
+metadata:
+  annotations:
+    openshift.io/image.insecureRepository: "true"
```
Comment (vrutkovs) [inline templates/osbs-buildroot-imagestream.yml.j2]: This should be parametrized as well - Fedora's repository is secure and some day ours will be too

Comment (rcerven) [reply templates/osbs-buildroot-imagestream.yml.j2]: ok

File: tasks/orchestrator.yml [hunk 7]
```diff
@@ -16,3 +16,24 @@
...
   - oc
+
+# Setup imagestream
+- name: copy imagestream
+  template:
+    src: osbs-buildroot-imagestream.yml.j2
+    dest: "{{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml"
+  environment: "{{ osbs_environment }}"
+  register: yaml_imagestream
+  tags:
```
Comment (twaugh) [inline tasks/orchestrator.yml]: Do we only need to do these steps when osbs_buildroot_repository is set?

Comment (rcerven) [reply tasks/orchestrator.yml]: ahh right, now that it is variable it makes sense

File: tasks/orchestrator.yml [hunk 8]
```diff
@@ -16,3 +16,25 @@
...
+  - oc
+
+- name: create imagestream
+  command: >
+    oc replace
+    --namespace={{ osbs_namespace }}
+    --force=true
+    --filename={{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml
+  environment: "{{ osbs_environment }}"
+  when: yaml_imagestream.changed
```
Comment (twaugh) [inline tasks/orchestrator.yml]: This might need to be `when: osbs_buildroot_repository != '' and yaml_imagestream.changed`, because I don't think the registered variable `yaml_imagestream` will exist if the previous step is not run. (will it?)

Comment (rcerven) [reply tasks/orchestrator.yml]: "If a task fails or is skipped, the variable still is registered with a failure or skipped status, the only way to avoid registering a variable is using tags." 
http://docs.ansible.com/ansible/latest/playbooks_variables.html

so it will exist
